### ISSUE TITLE
A device received a reboot command but did not actually reboot

### ISSUE THREAD

**Issue by Page-** [resin-io/resin-supervisor#392]
HOST OS VERSION
Resin OS 2.0.0-beta10.rev1

SUPERVISOR VERSION
3.0.0

<img src="https://frontapp.com/assets/img/icons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_14y3)

**Comment by Page-** [comment]
The logs show that the reboot was attempted
```
Feb 23 22:00:37 0cfc814 start-resin-supervisor[683]: [2017-02-23T22:00:37.542Z] Event: Reboot {}
Feb 23 22:00:37 0cfc814 start-resin-supervisor[683]: 2017/02/23 22:00:37 api.go:120: Rebooting
Feb 23 22:04:22 0cfc814 start-resin-supervisor[683]: [2017-02-23T22:04:22.201Z] Event: Update notification {}
```
but the uptime makes it clear that it never actually happened
```
uptime
22:08pm up 23:12, 1 user, load average: 2.93, 2.58, 2.64
```

**Comment by Page-** [comment]
The container is using systemd with the dbus address set, in case that affects things

**Comment by dfunckt** [comment]
Could it be that updates were locked? It seems that super won't restart/reboot with update lock in place: https://github.com/resin-io/resin-supervisor/issues/373

**Comment by pcarranzav** [comment]
The log from api.go indicating that it's rebooting means that updates were not locked:
```
func inASecond(theFunc func()) {
	time.Sleep(time.Duration(time.Second))
	theFunc()
}

//RebootHandler Reboots the device using Systemd
func RebootHandler(writer http.ResponseWriter, request *http.Request) {
	log.Println("Rebooting")

	sendResponse, sendError := responseSenders(writer)
	if systemd.Logind == nil {
		sendError(fmt.Errorf("Logind unavailable, cannot reboot."))
		return
	}
	sendResponse("OK", "", http.StatusAccepted)
	go inASecond(func() { systemd.Logind.Reboot(false) })
}
```
So if the device logged "Rebooting" but didn't reboot, it's either:
* logind was unreachable from the supervisor container or
* the Go supervisor crashed before being able to run the Reboot function

It's very weird that there's no error logs in any case.

**Comment by xoul** [comment]
I've faced the same problem when my RPi had troubles connecting to the Internet via a 3G USB dongle.

The app on the RPi was trying to getting in touch with a server of mine, but from time to time the 3G USB dongle was losing connection, so I decided to force a reboot via Supervisor every time I could not communicate with the server.

What I saw is that GET /ping worked, but POST /v1/reboot was apparently ignored.

So, localhost would be up and running, but apparently this is not sufficient for the reboot code to work.

**Comment by pcarranzav** [comment]
@xoul thanks for reporting, what supervisor version did you see this on?

**Comment by xoul** [comment]
@pcarranzav I was using Resin OS 2.0.8+rev1 and Supervisor 5.1.0.
Maybe I can try no arrange another test on another board in the next days.

**Comment by pcarranzav** [comment]
What we found is that the cause for this was https://github.com/resin-io-library/base-images/issues/336 - the container's systemd was "stepping on" the host's DBUS. This caused the reboot call to get lost.

Using newer base images should have this fixed, and it wouldn't really be a supervisor issue in any case - so I'm closing this issue.

Users facing this issue and being unable to update the base image should unset DBUS_SYSTEM_BUS_ADDRESS and if necessary just set it for the precise program that needs it (and not for the whole container). A manual reboot might be needed to get the device back into a fully working state.

(If you're seeing this issue and you never set DBUS_SYSTEM_BUS_ADDRESS, @xoul, let me know and I'll reopen - we'd need to look at your devices to figure out what's going on)

**Comment by xoul** [comment]
@pcarranzav I've updated the RPi3 remotely from Resin OS 2.0.8+rev1 and Supervisor 5.1.0 to Resin0S 2.3.0 and Supervisor 6.1.3 yesterday, but with the same results. 
I've not tried 2.7.5, yet. Do you that that the problem has been addressed in 2.7.5 only?

I will try to investigate further.
This RPi3 is connected to the Internet via a 3G USB stick. The connection is dropped sometimes.
When the RPi3 is connected I can reboot it via the Supervisor API.
When the RPI3 is not connected I apparently can't reboot it via the Supervisor API.

I will add back some logging via crontab and try to just remove the 3G USB stick.
I will log "ip a", "uptime" and something else. Then try to reboot.
